### ISSUE TITLE
calling Load(config) from modules

### ISSUE THREAD

**Issue by aep** [caddyserver/caddy#4464]
the website mentions dynamic configuration but i haven't found any plugin that dynamically loads configs, and no documentation.

Is it ok to call Load() from a module? https://github.com/caddyserver/caddy/blob/e7457b43e4703080ae8713ada798ce3e20b83690/caddy.go#L102

**Comment by francislavoie** [comment]
You're right, we haven't really documented how to use dynamic config loading. There's an explanation in the PR where it was introduced though, see the 3rd part https://github.com/caddyserver/caddy/pull/3994

**Comment by aep** [comment]
thanks. the linked PR doesnt explain if Load can be called at any point tho, as it is going to happen with a dynamic config.
Instead it seems to be intended for static configs loaded from a remote location that only gets loaded once.

**Comment by francislavoie** [comment]
Right, the other part is https://github.com/caddyserver/caddy/pull/4246 which added periodically pulling the config from the loader. So you could implement your own loader (a module in the namespace `caddy.config_loaders`).

**Comment by aep** [comment]
which would only allow polling instead of loading it on change tho.
i could poll every 1ms i guess, but then again i might as well just call the http/admin endpoint from itself, which seems to support reloading on demand.

**Comment by francislavoie** [comment]
> but then again i might as well just call the http/admin endpoint from itself

The issue with that is the risk of deadlocks, cause the thing making the request will try to get shut down by the config loader which is swapping out the old config with the new, but if it can't shut down the old config (because there's still a pending request), it'll never be able to load the new config.

**Comment by aep** [comment]
ah, hm this is awkward. I found some other issues with async loading but don't remember since we ended up not using Caddy.

I think we should just close this.
